src/components/CustomTable/CustomTable.tsx

```diff
@@ -0,0 +1,143 @@
+import { Box, Table, TableBody, TableCell, TableContainer, TablePagination, TableRow } from '@mui/material';
+import { ChangeEvent, MouseEvent, useEffect } from 'react';
+import Checkbox from '@mui/material/Checkbox';
+import { CustomTableHead } from '../CustomTableHead/CustomTableHead';
+import { CustomTableProps } from './CustomTable.types';
+import { FetchError } from '../../errors/FetchError';
+
+export function CustomTable<T extends Record<PropertyKey, unknown>>(props: CustomTableProps<T>) {
+  const { toolbar, fetchUrl, selectedRowsIds, page, rows, rowsPerPage, setSelectedRowsIds, setPage, setRowsPerPage, setRows } = props;
+
+  const isSelected = (row: T) => {
+    return (
+      selectedRowsIds.filter((item) => {
+        return (row.id as string) === item;
+      }).length > 0
+    );
+  };
+
+  const pixelHeightPerRow = 53;
+  const rowsPerPageArray = [5, 10, 25];
+
+  const handleClick = (event: MouseEvent<unknown>, row: T) => {
+    const selectedIndex = selectedRowsIds.indexOf(
+      selectedRowsIds.find((item) => {
+        return String(row.id) === item;
+      }) as string
```

Comment:
why `as string` ?

Reply:
Removed in the latest version.

Reply:
indexOf can return undefined
![image](https://github.com/dorm-social-room-manager/dsrm-frontend/assets/72614768/c022ce19-4aa4-49bb-8e7b-4be3e68c8f3a)

Reply:
so now you let pass undefined to `inedOf`. It is incorrect. You should handle scenario when `.find` returns `undefined`. + you can always use `.findIndex` :wink:

Reply:
fixed